src/studio/src/designer/frontend/packages/schema-editor/src/features/editor/schemaEditorSlice.ts

```diff
@@ -307,21 +301,33 @@ const schemaEditorSlice = createSlice({
     },
     setPropertyName(state, action) {
       const {
-        path, name, navigate,
+        path, navigate,
       } = action.payload;
+      let name = action.payload.name;
       if (!name || name.length === 0) {
         return;
       }
+
+      // make sure property name is unique
+      const [parentPath] = getParentPath(path);
+      if (parentPath != null) {
+        const parent = getUiSchemaItem(state.uiSchema, parentPath);
+        if (parent.properties) {
+          if (parent.properties?.findIndex((p) => p.displayName === name) > -1) {
```

Comment:
You are already testing `properties` on the line above.
You can also replace the query with `find((p) => p.displayName === name)`
Or you could be more explicit and use `!== -1`.